### PR TITLE
Extended scenario tests

### PR DESCRIPTION
**Pull request by wrinkledeth**
# New scenario tests

    test33333NodeRegister()
    test33NodeRegister()
    test53NodeQuit()
    test5NodeQuit()
    test63Group0NodeQuit()
    test63Group1NodeQuit()
    test66666NodeRegister()
    test66NodeRegister()
    testNodeSetup()

- Implemented 9 new scenario tests (98% test coverage reached)
- Uploaded detailed notes describing all test scenarios.
- Fixed a bug in findOrCreateTargetGroup() that was breaking multiple rebalancing scenarios

### CODE REVIEW

**Comment by wrinkledeth** [comment]
copilot:all

File: contracts/test/ExtendedDKGScenarioTests.t.sol [hunk 1]
```diff
@@ -0,0 +1,782 @@
...
+    }
+
+    function dkgHelper(Params[] memory params) public {
+        for (uint256 i = 0; i < params.length; i++) {
+            vm.prank(params[i].nodeIdAddress);
+            if (params[i].shouldRevert) {
+                vm.expectRevert(params[i].revertMessage);
+            }
+            controller.commitDkg(
+                IController.CommitDkgParams( // ! IController
```
Comment (kafeikui) [inline contracts/test/ExtendedDKGScenarioTests.t.sol:161]: Is it ok to move this?

File: contracts/test/ExtendedDKGScenarioTests.t.sol [hunk 2]
```diff
@@ -0,0 +1,782 @@
...
+
+        registerIndex(7); // added to group_0, only group_0 epoch is incremented
+        assertEq(controller.getGroup(0).epoch, 5); // g.epoch++
+        assertEq(controller.getGroup(1).epoch, 1); // no change
+        assertEq(controller.getGroup(0).size, 4); // g.size++
+        assertEq(controller.getGroup(1).size, 3);
+        assertEq(checkIsStrictlyMajorityConsensusReached(0), false);
+        assertEq(checkIsStrictlyMajorityConsensusReached(1), false);
+
+        registerIndex(8); // added to group_1, only group_1 epoch is incremented // ! problem here
```
Comment (kafeikui) [inline contracts/test/ExtendedDKGScenarioTests.t.sol:288]: The same above.

File: contracts/test/ExtendedDKGScenarioTests.t.sol (hunk 1 above)
Comment (wrinkledeth) [reply contracts/test/ExtendedDKGScenarioTests.t.sol:161]: fixed

File: contracts/test/ExtendedDKGScenarioTests.t.sol (hunk 2 above)
Comment (wrinkledeth) [reply contracts/test/ExtendedDKGScenarioTests.t.sol:288]: fixed

**Pull request merged**